### DISCUSSION TITLE
Should I practise on Codewars if I just start learning Python?

### DISCUSSION THREAD

**Discussion by BegginerDice** [codewars/codewars.com#2662] [Q&A] [answered]
I started learning programming 3 days ago (very actively), I am studying from the book "Python crash course" (E. Matthes). I also watch a training course on YouTube to consolidate my knowledge. I really like solving challenges on Codewars (6 kyu), but comparing my solution with others, I notice that sometimes I chose more complex solutions. So that's my question : Should I continue to borrow on Codewars, or at least finish the book first, so as not to get used to writing more complex code?
(sorry for the bad english)

**Comment by Stefan3002** [comment, marked as answer]
Hello there. My personal advice is to keep practicing coding problems on CodeWars, why not? You can solve the easier ones (8-7 kyu) to get more comfortable with Python's functions and other methods. Regarding other peoples solutions: don't worry too much about that, with time you will get there. The people that write those lines of code, are very very experienced, it will take years of practice to get there, but don't be discouraged, anything is possible if you put your mind into it. Take me for example: I am currently 4 kyu, very close to 3 kyu and I still don't know a lot of those methods to solve katas, but that is not an issue, I just try to understand what they did there, search some documentations and that's it. Also, a one line code is usually impressive, but it is not very readable, so it is not always the best solution. Lastly, remember that Python has almost every time a shorter way of doing things, hence the expression : "The Pythonic way of solving this is: ...", it always has some weird functions I have never even heard of. To summarize: sure, keep solving katas if that is what you wish! All the best! Also, remember this is just my personal opinion. 👍

**Reply by BegginerDice** [reply]
Thanks for the answer, I really like to think over the solution algorithm, and then Google new commands and methods if my knowledge is not enough. It's time to stop being embarrassed by my 5 - 15 lines of code instead of one line lambda function and keep trying to learn the list generator👾

**Comment by WarriorDXX** [comment]
lmao i started 3 days ago too ( very actively) currently 5 kyu at js and i cheated quite a bit because i was too lazy to code the solution i just somewhat solved it mentally and copied the code so 6 kyu actual level and im just focusing on js and reading the definitive guide on js and already finished some of the cs50 material and other courses SO you are not alone and trust me getting into codewars quickly and just solving tens of katas each day is the way to go no matter what the whole process you go through each time you solve a problem is very similar whether you are solving artificial or actual problems in real life projects and no NO never rely on books or courses or whatever that's good but no matter what you have to write some code and solve some problems because think about it if let's say programming books didn't exist and the only way to learn was with docs and practicing then you would still be a pretty good programmer maybe a bit extra difficult but theoretically books are kinda useless practicing is the way 

(deleted it by mistake)
